dinstaller-lib/src/settings.rs

```diff
@@ -0,0 +1,141 @@
+use crate::users::{UsersClient,FirstUser};
+use crate::storage::StorageClient;
+use std::{str::FromStr, error::Error};
+
+#[derive(Debug, Default)]
+pub struct Settings {
+    pub users: Option<UsersSettings>,
+    pub storage: Option<StorageSettings>
+}
+
+#[derive(Debug)]
+pub struct UsersSettings {
+    pub first_user: Option<FirstUser>
+}
+
+#[derive(Debug)]
+pub struct StorageSettings {
+
+}
+
+/// Settings storage
+///
+/// It is responsible for loading and storing the settings in the D-Bus service.
+pub struct Store<'a> {
+    storage_client: StorageClient<'a>,
+    users_client: UsersClient<'a>,
+}
+
+impl<'a> Store<'a> {
+    pub fn new() -> Result<Self, Box<dyn Error>> {
```

Comment:
NP: I am not sure if we need boxing out errors with all its disadvantage here as only error can be here zbus::Error, not?